src/clr-angular/utils/drag-and-drop/providers/draggable-snapshot.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Injectable} from "@angular/core";
+import {DomAdapter} from "../../dom-adapter/dom-adapter";
+import {ClrDragEvent} from "../interfaces/drag-event";
+
+// This service is used to capture the state of clrDraggable element
+// at a certain event and passes it to clrDraggableGhost component.
+@Injectable()
+export class ClrDraggableSnapshot<T> {
+    constructor(private domAdapter: DomAdapter) {}
+
+    private draggableElClientRect: ClientRect;
+    private stateEvent: ClrDragEvent<T>;
```

Comment:
shouldn't this be `dragEvent` instead of `stateEvent`? The naming seems off.

Reply:
this private property has become `snapshotEvent`, but gets accessed as `get dragEvent()` now.